### DISCUSSION TITLE
Set env variables for scheduled workflow based on cron expression

### DISCUSSION THREAD

**Discussion by andrzej-woof** [community/community#25928] [Actions]
<p>I want to reuse my workflow jobs scheduling them at various times of the day but with different input<br>
Basically something like this</p>
<pre><code class="lang-auto">on:
  schedule:
    - cron:  '0 0 * * *'
      env:
        MY_VAR: midnight_run
    - cron:  '0 6 * * *'
      env:
        MY_VAR: morning_run
</code></pre>
<p>but I assume this is not supported?</p>
<p>I can obviously run some custom command/script that will check current date and (assuming that I give it some tolerance in terms of exact minute) set the correct value of <code>MY_VAR</code> but it doesn’t feel like a good solution.</p>
<p>I’ll appreciate any ideas :slight_smile:</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/andrzej-kodify">@andrzej-kodify</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>You cannot define top level env in the event, it’s not supported. As an alternative, based on the different schedule cron settting, you can use ‘<a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable" rel="nofollow noopener">set-env</a>’ to  set/update env value for the job.</p>
<p>Code sample as below:</p>
<pre><code class="lang-auto">on:
  schedule:
    - cron:  '*/6 * * * *'   # define cron setting
    - cron:  '*/7 * * * *'
jobs:
  jobenv:
    runs-on: ubuntu-latest
    steps:
      - name: setenv
        if: github.event.schedule=='*/6 * * * *'         # set-env value
        run: |
          echo "::set-env name=cronenv::midnight_run"
      - name: setenv2
        if: github.event.schedule=='*/7 * * * *'   # set env value
        run: |
          echo "::set-env name=cronenv::morning_run"
      - name: check env value  # check env value
        run: |
          echo $cronenv               
</code></pre>
<p>the var ‘cronenv’ has different value based on the different cron setting.<br>
Confirmed it works on my side: <a href="https://github.com/weide-zhou/schedulewf1/runs/921576254?check_suite_focus=true#step:6:6" rel="nofollow noopener">https://github.com/weide-zhou/schedulewf1/runs/921576254?check_suite_focus=true#step:6:6</a></p>
<p>Thanks</p>

**Reply by gokultw** [reply]
I was duplicating the same workflows these days, Thanks much @weide-zhou 👍

**Reply by tzachshabtay** [reply]
@weide-zhou sure would be nice to have a way to add data to a cron like a name so we wouldn't need to use the cron expression as an id (and be able to have multiple different crons with the same expression), i.e something like:
```
on:
  schedule:
    - cron:  '*/6 * * * *'
      name: 'service1'
    - cron:  '*/6 * * * *'
      name: 'service2'
jobs:
  test_service:
    runs-on: ubuntu-latest
    steps:
      - name: test
        run: make -C src/monolith/services/${{ github.event.schedule.name }} test
```

**Comment by andrzej-woof** [comment]
<p>Great, that’s the workaround I’ve been after. Thanks a lot :slight_smile:</p>

**Comment by Hobart2967** [comment]
Any news on this? This would be very helpful to us, too. Schedules may change over time, names or IDs dont.

**Comment by rajendran-k** [comment]
Any updates on this? At least any support for adding a name or id to cron job?

**Comment by Hobart2967** [comment]
Seems like we need to reopen this as it's marked as answered.
